src/renderer/components/dock/create-resource.tsx

```diff
@@ -21,6 +25,39 @@ interface Props {
 @observer
 export class CreateResource extends React.Component<Props> {
   @observable error = "";
+  templates:GroupSelectOption[] = [];
+
+  componentDidMount(){
+    let rootPaths: any[]= [];
+
+    for (const store of this.templatesStores) {
+      rootPaths = rootPaths.concat(fs.readdirSync(store).map(value => {return {root: store, value};}));
+    }
+    const ungroupedFiles: any[] = [];
+
+    rootPaths.forEach((rootPath, index) => {
+      const absRootPath = path.join(rootPath.root, rootPath.value);
+
+      const isOverrided :boolean = rootPaths.map(v => v.value).indexOf(rootPath.value) < index;
+
+      if (!isOverrided){
+        if (fs.statSync(absRootPath).isDirectory()) {
+          const files = fs.readdirSync(absRootPath).map(file => {
+            return {value: path.join(absRootPath, file), label: path.parse(file).name};
+          });
+
+          this.templates.push({label: rootPath.value.replace("-", " "), options: files});
+        }
+        else {
+          ungroupedFiles.push({value: absRootPath, label: path.parse(absRootPath).name});
+        }
+      }
+    });
+
+    if (ungroupedFiles.length > 0) {
+      this.templates.push({label: "Ungrouped", options: ungroupedFiles});
+    }
```

Comment:
Better to move all these logic to some named method with clear intention. Maybe `loadTemplates()`?